app/app.gradle.kts

```diff
@@ -65,6 +65,70 @@ tasks.withType<Test> {
     dependsOn(project("test_plugins").tasks["jar"])
 }
 
+val nativeShadowTasks = NativePlatforms.values().map { platform ->
+    tasks.create<ShadowJar>("shadowJar-${platform.platformName}") {
+        classifier = platform.platformName
+        configurations = listOf(
+                project.configurations.compile,
+                project.configurations.getByName(platform.platformName)
+        )
+        from(
+                java.sourceSets["main"].output,
+                project(":api").java.sourceSets["main"].output,
+                project(":plugins:base").java.sourceSets["main"].output,
+                project(":plugins:cameraserver").java.sourceSets["main"].output,
+                project(":plugins:networktables").java.sourceSets["main"].output,
+                project(":plugins:powerup").java.sourceSets["main"].output
```

Comment:
You could probably make this more generic and less copy pasted.

Reply:
I wish.  I was trying to get it into the `nativeProject` function but the DependencyHandler class doesn't have access to the enclosing project

Reply:
```kotlin
val projectOutputs = 
     setOf(project(":api"), project(":plugins:base"), ....)
         .map { it.java.sourceSets["main"].output  } 
val outputs =
    setOf(java.sourceSets["main"].output) + projectOutputs
from(outputs)
```